### PR TITLE
b2b Opportunities changes

### PR DESCRIPTION
**Pull request by prabhum2**
Please link to the issue #…

### CODE REVIEW

File: components/classes/opportunity-person.schema.json [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      "$ref": "https://ns.adobe.com/xdm/data/record"
+    },
+    {
+      "$ref": "#/definitions/opportunity-person"
+    },
+    {
+      "$ref": "https://ns.adobe.com/xdm/common/external-source-system-audit-details"
+    }
+  ],
+  "required": ["xdm:opportunityID"],
```
Comment (cmathis) [inline components/classes/opportunity-person.schema.json:42]: Where is this field defined?

File: components/mixins/opportunity/opportunities-details.schema.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  },
+  "allOf": [
+    {
+      "$ref": "https://ns.adobe.com/xdm/common/extensible#/definitions/@context"
+    },
+    {
+      "$ref": "#/definitions/opportunity-details"
+    },
+    {
+      "$ref": "https://ns.adobe.com/xdm/datatypes/b2b-opportunity"
```
Comment (cmathis) [inline components/mixins/opportunity/opportunities-details.schema.json:29]: Why a B2B-Opportunity datatype and not just having those fields be defined in this mixin? When is the datatype to be used outside of this mixin?

File: components/mixins/opportunity/opportunity-person-relation.schema.json [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  },
+  "allOf": [
+    {
+      "$ref": "https://ns.adobe.com/xdm/common/extensible#/definitions/@context"
+    },
+    {
+      "$ref": "#/definitions/opportunity-person-relation"
+    },
+    {
+      "$ref": "https://ns.adobe.com/xdm/datatypes/b2b-opportunity-person"
```
Comment (cmathis) [inline components/mixins/opportunity/opportunity-person-relation.schema.json:29]: Same comment here. Why a b2b-opportunity-person datatype and not just fields defined in this mixin?

Also, should these opportunity->person mixins really be a generic global schema instead? Are these ever intended to be extended by the user?

File: components/classes/opportunity-person.schema.json [hunk 4]
```diff
@@ -0,0 +1,44 @@
+{
+  "meta:license": [
+    "Copyright 2020 Adobe Systems Incorporated. All rights reserved.",
+    "This work is licensed under a Creative Commons Attribution 4.0 International (CC BY 4.0) license",
+    "you may not use this file except in compliance with the License. You may obtain a copy",
+    "of the License at https://creativecommons.org/licenses/by/4.0/"
+  ],
+  "$id": "https://ns.adobe.com/xdm/classes/opportunity-person",
+  "$schema": "http://json-schema.org/draft-06/schema#",
+  "title": "XDM Business Opportunity Person Relation",
```
Comment (mjasrotia) [inline components/classes/opportunity-person.schema.json:10]: Should we remove the relation from the title ?

**Review by cmathis — APPROVED** [review]

File: components/mixins/opportunity/opportunity-person-components.schema.json [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+  "$id": "https://ns.adobe.com/xdm/mixins/opportunity-person-components",
+  "$schema": "http://json-schema.org/draft-06/schema#",
+  "title": "Opportunity Person Relation Components",
+  "type": "object",
+  "meta:extensible": true,
+  "meta:abstract": true,
+  "meta:intendedToExtend": [
+    "https://ns.adobe.com/xdm/classes/opportunity-person"
+  ],
+  "description": "XDM business opportunity person relation component details.",
```
Comment (cmathis) [inline components/mixins/opportunity/opportunity-person-components.schema.json:17]: This description does not tell me when to use this mixin. Please describe the use-case when this mixin is used.

File: components/mixins/opportunity/opportunity-person-details.schema.json [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+  "$id": "https://ns.adobe.com/xdm/mixins/opportunity-person-details",
+  "$schema": "http://json-schema.org/draft-06/schema#",
+  "title": "opportunity Person Relation Details",
+  "type": "object",
+  "meta:extensible": true,
+  "meta:abstract": true,
+  "meta:intendedToExtend": [
+    "https://ns.adobe.com/xdm/classes/opportunity-person"
+  ],
+  "description": "XDM business opportunity person relation details.",
```
Comment (cmathis) [inline components/mixins/opportunity/opportunity-person-details.schema.json:17]: Please update the description so it describes what use-case should use this mixin.
If you have no specific use-case, the move these fields to the class.

File: components/classes/opportunity-person.schema.json [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+  "type": "object",
+  "auditable": true,
+  "meta:extensible": true,
+  "meta:abstract": true,
+  "meta:extends": ["https://ns.adobe.com/xdm/data/record"],
+  "description": "An XDM Business Opportunity Person Relation.",
```
Comment (cmathis) [inline components/classes/opportunity-person.schema.json:16]: Is this class ever used without one of the two mixins you have defined? 
Does it ever makes sense to have the schema reference this class and not the oppertunity-person-details mixin also in this PR?

File: components/mixins/opportunity/opportunity-person-components.schema.json [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+          "type": "array",
+          "items": {
+            "type": "object",
+            "properties": {
+              "xdm:sourceOpportunityPersonID": {
+                "title": "Source Opportunity Person ID",
+                "description": "Opportunity person relationship unique identifier at the source.",
+                "$ref": "https://ns.adobe.com/xdm/datatypes/b2b-source"
+              },
+              "xdm:sourceExternalID": {
```
Comment (mjasrotia) [inline components/mixins/opportunity/opportunity-person-components.schema.json:31]: I don't think we need this property here as its not existing in the "opportunity-person-details". If its required then we will have to add it to the details mixin as well.

**Comment by peichiang** [comment]
Looks good to me. Thank you, Prabhakar!

**Review by cmathis — APPROVED** [review]

**Pull request merged**